core/src/wsv.rs

```diff
@@ -279,23 +279,155 @@ impl WorldStateView {
+
+                            dangerous_reference.pipeline_triggers.get(&id).map(|entry| {
+                                let action = entry.value();
+                                if let Repeats::Exactly(atomic) = action.repeats() {
+                                    if atomic.get() == 0 {
+                                        return;
+                                    }
+                                }
+
+                                match self.process_trigger(action, event_type) {
+                                    Ok(()) => succeed.push(id),
+                                    Err(err) => errors.push(err),
+                                }
+                            })
+                        }
+                        Event::Time(_) => {
+                            let dangerous_reference = unsafe {
+                                let ptr = &self.world.triggers as *const TriggerSet; // TODO: REALLY FIX THIS
+                                &*ptr
+                            };
+
+                            dangerous_reference.time_triggers.get(&id).map(|entry| {
+                                let action = entry.value();
+                                if let Repeats::Exactly(atomic) = action.repeats() {
+                                    if atomic.get() == 0 {
+                                        return;
+                                    }
+                                }
+
+                                match self.process_trigger(action, event_type) {
+                                    Ok(()) => succeed.push(id),
+                                    Err(err) => errors.push(err),
+                                }
+                            })
+                        }
+                        Event::ExecuteTrigger(_) => {
+                            let dangerous_reference = unsafe {
+                                let ptr = &self.world.triggers as *const TriggerSet; // TODO: REALLY FIX THIS
+                                &*ptr
+                            };
+
+                            dangerous_reference.by_call_triggers.get(&id).map(|entry| {
+                                let action = entry.value();
+                                if let Repeats::Exactly(atomic) = action.repeats() {
+                                    if atomic.get() == 0 {
+                                        return;
+                                    }
+                                }
+
+                                match self.process_trigger(action, event_type) {
+                                    Ok(()) => succeed.push(id),
+                                    Err(err) => errors.push(err),
+                                }
+                            })
+                        }
+                    };
+                }
+
+                if errors.is_empty() {
+                    (succeed, Ok(()))
+                } else {
+                    (succeed, Err(errors))
+                }
```

Comment:
```suggestion
(succeed, if errors.is_empty() {Ok(())} else {Err(errors)})
```
Though if we think about the function it could be done even more compactly.

Reply:
I think this change is less readable, even if it is more compact. The function ought to be rewritted at some point though. I think the whole 'inspect with function' pattern is very harmful.

Reply:
Because mutating the thing which you are inspecting is technically a immutable and mutable borrow at the same time. It works with interior mutability, but that means that you have locks causing it to be a inherently single threaded task. I have no problem with the 'inspect with function' as long as we don't have any interior mutability. That way you can collect a set of changes to make and then apply them after you've inspected. This makes more sense in the context of a patch based world state view.

Reply:
Resolved with rollback.